query/common/src/main/java/org/codice/ddf/admin/common/report/ReportImpl.java

```diff
@@ -15,64 +15,79 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
-import org.apache.commons.collections.ListUtils;
 import org.codice.ddf.admin.api.report.ErrorMessage;
 import org.codice.ddf.admin.api.report.Report;
 
 import com.google.common.collect.ImmutableList;
 
-public class ReportImpl implements Report {
+public class ReportImpl<T> implements Report<T> {
 
-    private List<ErrorMessage> argumentMessages;
+    private List<ErrorMessage> errorMessages;
 
-    private List<ErrorMessage> resultMessages;
+    private Optional<T> result;
 
     public ReportImpl() {
-        argumentMessages = new ArrayList<>();
-        resultMessages = new ArrayList<>();
+        errorMessages = new ArrayList<>();
+        this.result = Optional.empty();
     }
 
-    @Override
-    public List<ErrorMessage> argumentMessages() {
-        return ImmutableList.copyOf(argumentMessages);
+    public ReportImpl(T result) {
+        this();
+        this.result = Optional.of(result);
     }
 
-    @Override
-    public List<ErrorMessage> resultMessages() {
-        return ImmutableList.copyOf(resultMessages);
+    public ReportImpl(ErrorMessage message) {
+        this();
+        errorMessages.add(message);
     }
 
-    @Override
-    public List<ErrorMessage> messages() {
-        return ListUtils.union(argumentMessages, resultMessages);
+    public ReportImpl(List<ErrorMessage> messages) {
+        this();
+        errorMessages.addAll(messages);
     }
 
-    public ReportImpl addArgumentMessages(List<ErrorMessage> messages) {
-        argumentMessages.addAll(messages);
-        return this;
+    public ReportImpl(Report<Object> report) {
```

Comment:
I don't think this constructor is very clear. It's only adding the messages and not the result. Is anyone using this? If not, I say we remove it.